### ISSUE TITLE
Differentiate satisfaction criteria arguments from activity creation arguments

### ISSUE THREAD

**Issue by camargo** [NASA-AMMOS/aerie#442]
For scheduling goals the planners should be able to edit some parameters without affecting the satisfaction criteria. `value` vs. `initial_value` syntax can be a nice solution here.

**Comment by adrienmaillard** [comment]
Posting here what Matt, Joel and I discussed on slack
```
Goal.CoexistenceGoal({
    activityTemplate: span => ActivityTemplates.BakeBananaBread({
      required: 5, // Must equal 5
      tbSugar: TODOMAGICWORD({
        "createWith": 5, 
        "satisfactionCriterion": $ => $.lessThan(10)
      }), // No restriction
    })
  }).mustSatisfy(instance => instance.tbSugar.equals(5))
```

**Comment by mattdailis** [comment]
Closed by #847
